### PR TITLE
Move OneWire to `onewireio`

### PR DESCRIPTION
**Pull request by tannewt**
It is documented as if it's only in `onewireio` but is on `busio` and `bitbangio` as well for 7.x.

```
Adafruit CircuitPython 7.0.0-alpha.6-23-gd294692c4-dirty on 2021-08-12; Adafruit Metro M4 Express with samd51j19
>>> import busio
>>> dir(busio)
['__class__', '__name__', 'I2C', 'OneWire', 'SPI', 'UART']
>>> import bitbangio
>>> dir(bitbangio)
['__class__', '__name__', 'I2C', 'OneWire', 'SPI']
>>> import onewireio
>>> dir(onewireio)
['__class__', '__name__', 'OneWire']
```

### CODE REVIEW

File: py/circuitpy_mpconfig.h [hunk 1]
```diff
@@ -583,6 +583,13 @@ extern const struct _mp_obj_module_t neopixel_write_module;
 extern const struct _mp_obj_module_t nvm_module;
 #endif
 
+#if CIRCUITPY_OS
```
Comment (jepler) [inline py/circuitpy_mpconfig.h:586]: CIRCUITPY_ONEWIREIO?

**Review by jepler — CHANGES_REQUESTED** [review]

**Review by dhalbert — APPROVED** [review]
Looks good!

**Comment by cwalther** [comment]
Is it intended that you can’t set `CIRCUITPY_ONEWIREIO = 0` in 7.0 yet because `busio` and `bitbangio` still unconditionally refer to it?

(I was just wondering if doing that might let us enable `pwmio` on the PewPew M4 instead – PWM strikes me as a more useful thing to have than than OneWire.)

**Comment by tannewt** [comment]
I hadn't considered making onewire removable in 7.0.0. I'm open to it on a board-by-board basis but we'd need to make it work. :-)

**Comment by cwalther** [comment]
That is easy, unless I’m missing anything:
```diff
diff --git a/shared-bindings/bitbangio/__init__.c b/shared-bindings/bitbangio/__init__.c
index a8d9f5e61..acf7f05e4 100644
--- a/shared-bindings/bitbangio/__init__.c
+++ b/shared-bindings/bitbangio/__init__.c
@@ -72,7 +72,9 @@
 STATIC const mp_rom_map_elem_t bitbangio_module_globals_table[] = {
     { MP_ROM_QSTR(MP_QSTR___name__), MP_ROM_QSTR(MP_QSTR_bitbangio) },
     { MP_ROM_QSTR(MP_QSTR_I2C),   MP_ROM_PTR(&bitbangio_i2c_type) },
+    #if CIRCUITPY_ONEWIREIO
     { MP_ROM_QSTR(MP_QSTR_OneWire),   MP_ROM_PTR(&onewireio_onewire_type) },
+    #endif
     { MP_ROM_QSTR(MP_QSTR_SPI),   MP_ROM_PTR(&bitbangio_spi_type) },
 };
 
diff --git a/shared-bindings/busio/__init__.c b/shared-bindings/busio/__init__.c
index 2cde3d62b..38efd06bd 100644
--- a/shared-bindings/busio/__init__.c
+++ b/shared-bindings/busio/__init__.c
@@ -73,7 +73,9 @@ STATIC const mp_rom_map_elem_t busio_module_globals_table[] = {
     { MP_ROM_QSTR(MP_QSTR___name__), MP_ROM_QSTR(MP_QSTR_busio) },
     { MP_ROM_QSTR(MP_QSTR_I2C),   MP_ROM_PTR(&busio_i2c_type) },
     { MP_ROM_QSTR(MP_QSTR_SPI),   MP_ROM_PTR(&busio_spi_type) },
+    #if CIRCUITPY_ONEWIREIO
     { MP_ROM_QSTR(MP_QSTR_OneWire),   MP_ROM_PTR(&onewireio_onewire_type) },
+    #endif
     { MP_ROM_QSTR(MP_QSTR_UART),   MP_ROM_PTR(&busio_uart_type) },
 };
 
```

Shall I make a PR with that (even though I don’t have an immediate use for it after all)?

**Comment by tannewt** [comment]
> Shall I make a PR with that (even though I don’t have an immediate use for it after all)?

Sure! Always good to have the option.

**Pull request merged**